source/graphql/graphql.d

```diff
@@ -143,77 +143,64 @@ class GraphQLD(T, QContext = DefaultContext) {
 		}
 	}
 
-	Json getDefaultArgumentImpl(string typename, Type)(string type,
-			string field)
+	static Json getDefaultArgumentImpl(Type)(string field)
 	{
 		static if(isAggregateType!Type) {
-			if(typename == type) {
-				switch(field) {
-					static foreach(mem; __traits(allMembers, Type)) {
-						static if(isCallable!(
-								__traits(getMember, Type, mem))
-							)
-						{
-							case mem: {
-								alias parNames = ParameterIdentifierTuple!(
-										__traits(getMember, Type, mem)
-									);
-								alias parDef = ParameterDefaultValueTuple!(
-										__traits(getMember, Type, mem)
-									);
-
-								Json ret = Json.emptyObject();
-								static foreach(i; 0 .. parNames.length) {
-									static if(!is(parDef[i] == void)) {
-										ret[parNames[i]] =
-											serializeToJson(parDef[i]);
-									}
+			switch(field) {
+				static foreach(mem; __traits(allMembers, Type)) {
+					static if(isCallable!(
+							__traits(getMember, Type, mem))
+						)
+					{
+						case mem: {
+							alias parNames = ParameterIdentifierTuple!(
+									__traits(getMember, Type, mem)
+								);
+							alias parDef = ParameterDefaultValueTuple!(
+									__traits(getMember, Type, mem)
+								);
+
+							Json ret = Json.emptyObject();
+							static foreach(i; 0 .. parNames.length) {
+								static if(!is(parDef[i] == void)) {
+									ret[parNames[i]] =
+										serializeToJson(parDef[i]);
 								}
-								return ret;
 							}
+							return ret;
 						}
 					}
-					default: break;
 				}
+				default: break;
 			}
 		}
 		return Json.init;
 	}
 
 	Json getDefaultArguments(string type, string field) {
-		import graphql.traits : collectTypes;
-		switch(type) {
-			static foreach(Type; collectTypes!(T)) {{
-				case Type.stringof: {
-					Json tmp = getDefaultArgumentImpl!(Type.stringof, Type)(
-							type, field
-						);
-					if(tmp.type != Json.Type.undefined
-							&& tmp.type != Json.Type.null_)
-					{
-						return tmp;
-					}
-				}
-			}}
-			default: {}
+		import graphql.traits : execForAllTypes;
+		alias defaultArgFn = Json function(string) @safe;
+		static defaultArgFn[string] items;
```

Comment:
I might rework this function anyway. It seems kind of odd to generate these closures inside a method, when they can just be generated on the object itself, especially since it's a class. Less to allocate.

Reply:
fair enough

Reply:
Hm... I was incorrect, I was thinking of the setDefaultResolver function, which is outside the GraphQLD type.

I did move this out into the class, but my comment about generating closures inside a method was referring to the other function, which needs to keep allocating closures. However, it could easily be reworked into a class, which might actually make more sense as there may be more allocations than is necessary. A class would only have one allocation.